model/common/src/icon4py/model/common/model_options.py

```diff
@@ -13,18 +13,43 @@
     Scalar,
     is_scalar_type,  # TODO(havogt): Should this function be public API?
 )
-from gt4py.next import Field, backend
+from gt4py.next import Field
 from gt4py.next.common import OffsetProvider
 from gt4py.next.ffront.decorator import Program
 
+from icon4py.model.common.model_backends import make_custom_dace_backend, make_custom_gtfn_backend
+
 
 def dict_values_to_list(d: dict[str, typing.Any]) -> dict[str, list]:
     return {k: [v] for k, v in d.items()}
 
 
+def get_options(program_name: str, arch: str, **backend):
+    backend_kind = backend.get("backend_kind", "default")
+    return dict(backend_kind=backend_kind)
+
+
+def customize_backend(program: str = "", arch: str = "", **backend):
+    program_id = str(program.past_stage.past_node.id) if program != "" else program
+    options = get_options(program_id, arch, **backend)
+    if options["backend_kind"] == "dace":
+        backend_func = make_custom_dace_backend
+    elif options["backend_kind"] == "gtfn":
+        backend_func = make_custom_gtfn_backend
+    on_gpu = backend["device"] == "gpu"
+    custom_backend = backend_func(
+        on_gpu=on_gpu,
+        **options,
+    )
+    return custom_backend
```

Comment:
I think we should change the makers to accept the device kind that we will introduce instead of processing and translating to `on_gpu`. That would also allow a backend maker to reject invalid devices.

Reply:
Yes